pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -241,6 +242,13 @@ public Consumer subscribe(String topic, String subscription, ConsumerConfigurati
                     new PulsarClientException.InvalidConfigurationException("Consumer configuration undefined"));
         }
 
+        if (conf.getActiveConsumerListener() != null
```

Comment:
I know it doesn't make a lot of sense to check for active/inactive on a non failover subscription. But wouldn't be in a way still semantically correct to always notify of "active"  in exclusive/shared mode?

Reply:
I think the main motivation for this change here is for supporting failover subscription. I would disable this feature for other subscriptions for now, and introduce it if we really need it. does that make sense to you?